WordPress/Classes/Services/PostRepository.swift

```diff
@@ -63,6 +58,58 @@ final class PostRepository {
         }
     }
 
+    /// Uploads the changes to the given post to the server and deletes the
+    /// uploaded local revision afterward. If the post has an associated
+    /// remote ID, creates a new post.
+    ///
+    /// - note: This method is a low-level primitive for syncing the latest
+    /// post date to the server.
+    ///
+    /// - warning: Before publishing, ensure that the media for the post got
+    /// uploaded. Managing media is not the responsibility of `PostRepository.`
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    @MainActor
+    func _upload(_ post: AbstractPost) async throws {
+        let remotePost = PostHelper.remotePost(with: post)
+        try await _upload(remotePost, for: post)
+    }
+
+    /// Uploads the changes to the given post to the server and deletes the
+    /// uploaded local revision afterward. If the post has an associated
+    /// remote ID, creates a new post.
```

Comment:
tiny nit - I think there's a typo
```suggestion
    /// uploaded local revision afterward. If the post doesn't have an associated
    /// remote ID, creates a new post.
```